### PR TITLE
Better permissions for auth_token

### PR DESCRIPTION
**Pull request by arbll**
### What does this PR do?

Change the permissions of the `auth_token` file to `0600`. 

### Motivation

Security reasons.

### Additional Notes

This change will prevent users that can't read the `auth_token` from running any `datadog-agent` commands that need to talk to the running agent endpoint. While this make sense IMO, it remove some ease of use. On linux the user will have to `su dd-agent`.

### CODE REVIEW

File: pkg/api/security/platform_nowindows.go [hunk 1]
```diff
@@ -9,16 +9,9 @@ package security
...
 
-// writes auth token(s) to a file with the same permissions as datadog.yaml
+// writes auth token(s) to a file that is only readable by the user running the agent
 func saveAuthToken(token, tokenPath string) error {
-	confFile, err := os.Stat(config.Datadog.ConfigFileUsed())
-	if err != nil {
-		return err
-	}
-	return ioutil.WriteFile(tokenPath, []byte(token), confFile.Mode())
+	return ioutil.WriteFile(tokenPath, []byte(token), 0600)
```
Comment (olivielpeau) [inline pkg/api/security/platform_nowindows.go]: would it make sense to do the same thing on Windows? (not sure how permissions work on that file on windows)

File: pkg/api/security/platform_nowindows.go [hunk 2]
```diff
@@ -9,16 +9,9 @@ package security
 
 import (
 	"io/ioutil"
-	"os"
-
-	"github.com/DataDog/datadog-agent/pkg/config"
 )
 
-// writes auth token(s) to a file with the same permissions as datadog.yaml
+// writes auth token(s) to a file that is only readable by the user running the agent
```
Comment (olivielpeau) [inline pkg/api/security/platform_nowindows.go]: we should say `readable/writable` I think

**Review by olivielpeau — COMMENTED** [review]
Left a comment. Can you also add a release note? (this should be a change under `other` I think, it's not a security issue per se)

File: pkg/api/security/platform_nowindows.go (hunk 1 above)
Comment (arbll) [reply pkg/api/security/platform_nowindows.go]: Windows is already handled correctly : https://github.com/DataDog/datadog-agent/blob/eda9fe5fe0fa5591bc80e203a600f93b08ff61db/pkg/api/security/platform_windows.go#L34-L45

**Review by olivielpeau — APPROVED** [review]
LGTM 👍 

let's wait for the CI to pass and merge

**Comment by arbll** [comment]
Only fail is on `dogstatsd`, merging...

**Pull request merged**